pkgs/collection/lib/src/boollist.dart

```diff
@@ -10,7 +10,7 @@ import 'unmodifiable_wrappers.dart' show NonGrowableListMixin;
 /// A space-efficient list of boolean values.
 ///
 /// Uses list of integers as internal storage to reduce memory usage.
-abstract /*mixin*/ class BoolList with ListMixin<bool> {
+abstract interface class BoolList with ListMixin<bool> {
```

Comment:
I'd make it `final`. (Could be `sealed`, but nobody should care that there are exactly two implementations.)

There is zero reason to ever `implements BoolList`, it has the same interface as `List<bool>`.
The fact that you *can* shouldn't be taken as encouragement.

(We don't have `@Deprecate.interface()` yet, so there is no good way to deprecate implementability.)

Reply:
Went with `@sealed` for now, with a note about future breaking change.